ckanext/harvest/logic/schema.py

```diff
@@ -12,39 +13,48 @@
                                    )
 from ckan.logic.converters import convert_to_extras, convert_from_extras
 
-from ckan.lib.navl.validators import (ignore_missing,
-                                      not_empty,
-                                      ignore,
-                                      if_empty_same_as,
-                                      )
-
-from ckanext.harvest.logic.validators import (harvest_source_url_validator,
-                                              harvest_source_type_exists,
-                                              harvest_source_config_validator,
-                                              harvest_source_extra_validator,
-                                              harvest_source_frequency_exists,
-                                              dataset_type_exists,
-                                              harvest_source_convert_from_config,
-                                              harvest_source_id_exists,
-                                              harvest_job_exists,
-                                              harvest_object_extras_validator,
-                                              )
+from ckanext.harvest.logic.validators import (
+    harvest_source_url_validator,
+    harvest_source_type_exists,
+    harvest_source_config_validator,
+    harvest_source_extra_validator,
+    harvest_source_frequency_exists,
+    dataset_type_exists,
+    harvest_source_convert_from_config,
+    harvest_source_id_exists,
+    harvest_job_exists,
+    harvest_object_extras_validator,
+)
+ignore_missing = tk.get_validator("ignore_missing")
+not_empty = tk.get_validator("not_empty")
+ignore = tk.get_validator("ignore")
+if_empty_same_as = tk.get_validator("if_empty_same_as")
+
+ckan_version = int(h.ckan_version().split(".")[1])
+
+
+def unicode_safe(value):
+    if ckan_version >= 9:
+        from ckan.lib.navl.validators import unicode_safe as core_unicode_safe
+        return core_unicode_safe(value)
+    else:
+        return six.text_type(value)
```

Comment:
Sorry, I know that I suggested this but can we use `ckantoolkit.unicode_safe` here for better compatibility?
```suggestion
from ckantoolkit import unicode_safe
```
You will need to bump the ckantoolkit requirement to 0.0.6 in `pip-requirements.txt`